### PR TITLE
enpass-cli v1.6.0 (new formula)

### PR DESCRIPTION
**Pull request by kroussou**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/enpass-cli.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "./cmd/enpasscli"
+  end
+
+  test do
+    system bin/"enpass-cli", "version"
+    assert_match "enpass-cli arch=", shell_output("#{bin}/enpass-cli version 2>&1")
```
Comment (carlocab) [inline Formula/enpass-cli.rb:16]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Comment (kroussou) [reply Formula/enpass-cli.rb]: Thanks for your review! I've made some improvements to tests as you suggested.

File: Formula/enpass-cli.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    sha256 "a01d76ed4a2f3eaf6d695bb7727f80fb9f2b44f11f5850c1387c7ac6f45eadd2"
+  end
+  resource "testdata2" do
+    url "https://raw.githubusercontent.com/hazcod/enpass-cli/250b210987aa91ef3d556b1d93b22eca7fcf5c31/test/vault.enpassdb"
+    sha256 "451f7c6d7afcd6ea505d54a99a7b7a2c8ea79186f6c6c84daae73e2a6913c1c3"
+  end
+
+  def install
+    system "go", "build", *std_go_args, "./cmd/enpasscli"
+  end
```
Comment (carlocab) [inline Formula/enpass-cli.rb:21]:
```suggestion
    system "go", "build", *std_go_args, "./cmd/enpasscli"
    pkgshare.install "test/vault.json", "tesst/vault.enpassdb"
  end
```

File: Formula/enpass-cli.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  test do
+    # Download test vault
+    resource("testdata1").stage "#{testpath}/testvault/"
+    resource("testdata2").stage "#{testpath}/testvault/"
```
Comment (carlocab) [inline Formula/enpass-cli.rb:26]:
```suggestion
    mkdir "testvault"
    cp [pkgshare/"vault.json", pkgshare/"vault.enpassdb"], "testvault"
```

File: Formula/enpass-cli.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  resource "testdata1" do
+    url "https://raw.githubusercontent.com/hazcod/enpass-cli/250b210987aa91ef3d556b1d93b22eca7fcf5c31/test/vault.json"
+    sha256 "a01d76ed4a2f3eaf6d695bb7727f80fb9f2b44f11f5850c1387c7ac6f45eadd2"
+  end
+  resource "testdata2" do
+    url "https://raw.githubusercontent.com/hazcod/enpass-cli/250b210987aa91ef3d556b1d93b22eca7fcf5c31/test/vault.enpassdb"
+    sha256 "451f7c6d7afcd6ea505d54a99a7b7a2c8ea79186f6c6c84daae73e2a6913c1c3"
+  end
+
+  def install
```
Comment (carlocab) [inline Formula/enpass-cli.rb:19]:
```suggestion
  depends_on "go" => :build

  def install
```
If my suggestions above work then we don't need these.

**Comment by kroussou** [comment]
Your suggestion worked like a charm. Maybe it worth adding it to the [docs]?(https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md#:~:text=If%20your%20test%20requires%20a%20test%20file%20that%20isn%27t%20a%20standard%20test%20fixture%2C%20you%20can%20install%20it%20from%20a%20source%20repository%20during%20the%20test%20phase%20with%20a%20resource%20block%2C%20like%20this%3A)

File: Formula/enpass-cli.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  desc "Enpass command-line client"
+  homepage "https://github.com/hazcod/enpass-cli"
+  url "https://github.com/hazcod/enpass-cli/archive/refs/tags/v1.6.0.tar.gz"
+  sha256 "25668bb30747dc566695ff7f30e65fa2e29b04f5896155a4ab03185d8f5b4111"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "./cmd/enpasscli"
```
Comment (chenrui333) [inline Formula/enpass-cli.rb:11]:
```suggestion
    system "go", "build", *std_go_args(ldflags: "-s -w"), "./cmd/enpasscli"
```

**Comment by chenrui333** [comment]
One last thing, can you also assert the version? It should be good to go after that.

**Comment by kroussou** [comment]
@chenrui333 Thanks for review! I indeed forget to set version at build time. Fixed in [e53e18b](https://github.com/Homebrew/homebrew-core/pull/110897/commits/e53e18b2bfb32f69816527e246a6398bd49c2680)

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @kroussou!

**Comment by chenrui333** [comment]
@kroussou, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**